### PR TITLE
Changed L1 fee estimation logic for L2s

### PR DESCRIPTION
**Pull request by KuphJr**
To simplify the review, please use https://www.diffchecker.com/ to check the difference between the following files:
- `/v1_1_0/libraries/ChainSpecificUtil.sol` vs `/v1_3_1/libraries/ChainSpecificUtil.sol` [diff](https://www.diffchecker.com/7K5UxVGK/)
- `/v1_3_0/interfaces/IFunctionsBilling.sol` vs `/v1_3_1/interfaces/IFunctionsBilling.sol` [diff](https://www.diffchecker.com/eSIhMxqT/])
- `/v1_3_0/FunctionsBilling.sol` vs `/v1_3_1/FunctionsBilling.sol` [diff](https://www.diffchecker.com/HD6I21K8/)
- `/v1_3_0/FunctionsCoordinator.sol` vs `/v1_3_1/FunctionsCoordinator..sol` [diff](https://www.diffchecker.com/WuD6ZXZ8/)

Also, check diff between `/dev/v1_X` and /v1_3_1`

### CODE REVIEW

File: contracts/src/v0.8/functions/v1_1_0/libraries/ChainSpecificUtil.sol [hunk 1]
```diff
@@ -46,12 +45,14 @@ library ChainSpecificUtil {
   /// @notice On Arbitrum, the provided calldata is not used to calculate the fees.
   /// @notice On Optimism, the provided calldata is passed to the GasPriceOracle predeploy
   /// @notice and getL1Fee is called to get the fees.
-  function _getCurrentTxL1GasFees(bytes memory txCallData) internal view returns (uint256 l1FeeWei) {
+  function _getL1DataGasCostUpperLimit(bytes memory dataSizeBytes) internal view returns (uint256 l1FeeWei) {
```
Comment (justinkaseman) [inline contracts/src/v0.8/functions/v1_1_0/libraries/ChainSpecificUtil.sol]: `dataSizeBytes` implies it's a size number (which I agree is a better way), but the type is still bytes here.

File: contracts/src/v0.8/functions/v1_1_0/libraries/L1Block.sol [hunk 2]
```diff
@@ -0,0 +1,122 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.15;
+
+// Copied from the implementation contract for https://optimistic.etherscan.io/address/0x4200000000000000000000000000000000000015#readProxyContract
+// https://optimistic.etherscan.io/address/0x07dbe8500fc591d1852b76fee44d5a05e13097ff#code
+
+import { ISemver } from "./ISemver.sol";
```
Comment (justinkaseman) [inline contracts/src/v0.8/functions/v1_1_0/libraries/L1Block.sol:7]: If we want to do something like this, let's re-use the Chainlink style pattern using `ITypeAndVersion`
```
import {ITypeAndVersion} from "../../../shared/interfaces/ITypeAndVersion.sol";
```

File: contracts/src/v0.8/functions/v1_1_0/libraries/L1Block.sol [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+
+    /// @notice The scalar value applied to the L1 portion of the transaction fee.
+    /// @custom:legacy
+    uint256 public l1FeeScalar;
+
+    /// @notice The latest L1 blob base fee.
+    uint256 public blobBaseFee;
+
+    /// @custom:semver 1.2.0
+    string public constant version = "1.2.0";
```
Comment (justinkaseman) [inline contracts/src/v0.8/functions/v1_1_0/libraries/L1Block.sol:56]: If we want to do something like this, let's re-use the Chainlink style pattern using `ITypeAndVersion`
```
string public constant override typeAndVersion = "L1Block 1.0.0";
```

File: contracts/src/v0.8/functions/v1_1_0/libraries/L1Block.sol [hunk 4]
```diff
@@ -0,0 +1,122 @@
+// SPDX-License-Identifier: MIT
```
Comment (justinkaseman) [inline contracts/src/v0.8/functions/v1_1_0/libraries/L1Block.sol]: This is a vendored contract, not a library. Move this to `vendor`. 

In ChainSpecific we are calling the deployed contract, so really we only need an interface.
But the full contract may be useful as a test mock.

File: contracts/src/v0.8/functions/v1_1_0/FunctionsBilling.sol [hunk 5]
```diff
@@ -50,7 +50,8 @@ abstract contract FunctionsBilling is Routable, IFunctionsBilling {
     uint40 minimumEstimateGasPriceWei; //            ║ The lowest amount of wei that will be used as the tx.gasprice when estimating the cost to fulfill the request
     uint16 maxSupportedRequestDataVersion; // ═══════╝ The highest support request data version supported by the node. All lower versions should also be supported.
     uint224 fallbackNativePerUnitLink; // ═══════════╗ Fallback NATIVE CURRENCY / LINK conversion rate if the data feed is stale
-    uint32 requestTimeoutSeconds; // ════════════════╝ How many seconds it takes before we consider a request to be timed out
+    uint32 requestTimeoutSeconds; //                 ║ How many seconds it takes before we consider a request to be timed out
+    uint24 transmitTxSizeBytes; // ══════════════════╝ The size of the transmit transaction in bytes used to calculate the L1 data fee on L2s
```
Comment (justinkaseman) [inline contracts/src/v0.8/functions/v1_1_0/FunctionsBilling.sol]: To be more specific: this is the _expected maximum_ size (in bytes) to fulfill one request. Used to estimate the cost to post from L2 to L1.

File: contracts/src/v0.8/functions/v1_1_0/libraries/ChainSpecificUtil.sol (hunk 1 above)
Comment (justinkaseman) [inline contracts/src/v0.8/functions/v1_1_0/libraries/ChainSpecificUtil.sol]: Would it make sense to simplify this name to `_getL1FeeUpperLimit`?

File: contracts/src/v0.8/functions/v1_1_0/libraries/ChainSpecificUtil.sol [hunk 6]
```diff
@@ -46,12 +45,14 @@ library ChainSpecificUtil {
   /// @notice On Arbitrum, the provided calldata is not used to calculate the fees.
```
Comment (justinkaseman) [inline contracts/src/v0.8/functions/v1_1_0/libraries/ChainSpecificUtil.sol]: Function doc needs updating

File: contracts/src/v0.8/functions/v1_1_0/libraries/ChainSpecificUtil.sol [hunk 7]
```diff
@@ -46,12 +45,14 @@ library ChainSpecificUtil {
...
   /// @notice On Optimism, the provided calldata is passed to the GasPriceOracle predeploy
   /// @notice and getL1Fee is called to get the fees.
-  function _getCurrentTxL1GasFees(bytes memory txCallData) internal view returns (uint256 l1FeeWei) {
+  function _getL1DataGasCostUpperLimit(bytes memory dataSizeBytes) internal view returns (uint256 l1FeeWei) {
     uint256 chainid = block.chainid;
     if (_isArbitrumChainId(chainid)) {
-      return ARBGAS.getCurrentTxL1GasFees();
+      // https://docs.arbitrum.io/build-decentralized-apps/how-to-estimate-gas#where-do-we-get-all-this-information-from
```
Comment (justinkaseman) [inline contracts/src/v0.8/functions/v1_1_0/libraries/ChainSpecificUtil.sol]: Let's style this consistently with Optimism handling by either in-lining Optimism or giving Arbitrum a handler

**Comment by justinkaseman** [comment]
These changes should go into `dev/v1_X`, rather than the frozen version `/v1_1_0`

**Comment by KuphJr** [comment]
Closed in favor of this: https://github.com/smartcontractkit/chainlink/pull/13268

**Pull request closed without merging**